src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -473,32 +470,60 @@ private static ConfigurationItemFactory BuildDefaultFactory()
 #if !SILVERLIGHT && !NETSTANDARD1_3
         private static string GetAssemblyFileLocation(Assembly assembly)
         {
+            string fullName = string.Empty;
+
             try
             {
+                if (assembly == null)
+                {
+                    return string.Empty;
+                }
+
+                fullName = assembly.FullName;
+
+#if NETSTANDARD
+                if (string.IsNullOrEmpty(assembly.Location))
+                {
+                    // Assembly with no actual location should be skipped (Avoid PlatformNotSupportedException)
+                    InternalLogger.Warn("Skipping auto loading location because location is empty: ({1})", fullName);
+                    return string.Empty;
+                }
+#endif
+
                 Uri assemblyCodeBase;
                 if (!Uri.TryCreate(assembly.CodeBase, UriKind.RelativeOrAbsolute, out assemblyCodeBase))
                 {
-                    InternalLogger.Warn("No auto loading because assembly code base is unknown");
+                    InternalLogger.Warn("Skipping auto loading location because code base is unknown: `{0}` ({1})", assembly.CodeBase, fullName);
```

Comment:
not very important, but why the backticks and not quotes?

Reply:
Fixed